src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-share-form/smb-share-form.component.ts

```diff
@@ -0,0 +1,211 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { CdForm } from '~/app/shared/forms/cd-form';
+import { CdFormBuilder } from '~/app/shared/forms/cd-form-builder';
+import { CdFormGroup } from '~/app/shared/forms/cd-form-group';
+import { ActivatedRoute, Router } from '@angular/router';
+
+import _ from 'lodash';
+
+import { ActionLabelsI18n, URLVerbs } from '~/app/shared/constants/app.constants';
+import { FinishedTask } from '~/app/shared/models/finished-task';
+
+import { Filesystem, PROVIDER, SubVolume, SHARE_RESOURCE, SubVolumeGroup } from '../smb.model';
+import { DEFAULT_SUBVOLUME_GROUP } from '~/app/shared/constants/cephfs.constant';
+
+import { CephfsSubvolumeGroupService } from '~/app/shared/api/cephfs-subvolume-group.service';
+import { CephfsSubvolumeService } from '~/app/shared/api/cephfs-subvolume.service';
+import { SmbService } from '~/app/shared/api/smb.service';
+import { NfsService } from '~/app/shared/api/nfs.service';
+import { TaskWrapperService } from '~/app/shared/services/task-wrapper.service';
+import { map } from 'rxjs/operators';
+
+@Component({
+  selector: 'cd-smb-share-form',
+  templateUrl: './smb-share-form.component.html',
+  styleUrls: ['./smb-share-form.component.scss']
+})
+export class SmbShareFormComponent extends CdForm implements OnInit {
+  smbShareForm: CdFormGroup;
+  action: string;
+  resource: string;
+  allFsNames: Filesystem[] = [];
+  allsubvolgrps: SubVolumeGroup[] = [];
+  allsubvols: SubVolume[] = [];
+  defaultSubVolGroup = DEFAULT_SUBVOLUME_GROUP;
+  clusterId: string;
+
+  constructor(
+    private formBuilder: CdFormBuilder,
+    public smbService: SmbService,
+    public actionLabels: ActionLabelsI18n,
+    private nfsService: NfsService,
+    private subvolgrpService: CephfsSubvolumeGroupService,
+    private subvolService: CephfsSubvolumeService,
+    private taskWrapperService: TaskWrapperService,
+    private router: Router,
+    private route: ActivatedRoute
+  ) {
+    super();
+    this.resource = $localize`Share`;
+  }
+  ngOnInit() {
+    this.action = this.actionLabels.CREATE;
+    this.route.params.subscribe((params: { clusterId: string }) => {
+      this.clusterId = params.clusterId;
+    });
+    this.nfsService.filesystems().subscribe((data: Filesystem[]) => {
+      this.allFsNames = data;
+    });
+    this.createForm();
+  }
+
+  createForm() {
+    this.smbShareForm = this.formBuilder.group({
+      share_id: new FormControl('', {
+        validators: [Validators.required]
+      }),
+      volume: new FormControl('', {
+        validators: [Validators.required]
+      }),
+      subvolume_group: new FormControl(this.defaultSubVolGroup),
+      subvolume: new FormControl(''),
+      path: new FormControl({ value: '', disabled: true }),
+      path2: new FormControl('', {
```

Comment:
Instead of `path` and `path2` let's give them more descriptive names. Maybe something like:
`prefixedPath`, `inputPath`?